OutOfSchool/OutOfSchool.WebApi/Services/ChatRoomWorkshopService.cs

```diff
@@ -432,6 +434,22 @@ public async Task<ChatRoomWorkshopDto> GetUniqueChatRoomAsync(Guid workshopId, G
         }
     }
 
+    public async Task<int> GetCurrentUserUnreadMessagesCountAsync(Guid parentOrProviderId, Role userRole)
+    {
+        IEnumerable<ChatRoomWorkshopDtoWithLastMessage> chatrooms = null;
+        if (userRole == Role.Parent)
+        {
+            chatrooms = await GetByParentIdAsync(parentOrProviderId);
+        }
+
+        if (userRole == Role.Provider)
+        {
+            chatrooms = await GetByProviderIdAsync(parentOrProviderId);
+        }
+
+        return chatrooms.Count(chatroom => chatroom.NotReadByCurrentUserMessagesCount != 0);
```

Comment:
(Possible NRE) 
If `userRole` is neither `Role.Parent` nor `Role.Provider` - you'll get null set to `chatrooms` and thus `NullReferenceException` on `.Count()`. 
If it is a part of requirement (I'll left out of discussion downsides of the exeption-driven control flow approach), then please state it directly in code.
I.e. 
```
    public async Task<int> GetCurrentUserUnreadMessagesCountAsync(Guid parentOrProviderId, Role userRole)
    {
        var chatrooms = userRole switch {
            Role.Parent => await GetByParentIdAsync(parentOrProviderId),
            Role.Provider => await GetByProviderIdAsync(parentOrProviderId),
            _ => throw new ArgumentOutOfRangeException(nameof(userRole)),
        };

        return chatrooms.Count(chatroom => chatroom.NotReadByCurrentUserMessagesCount != 0);
    }
```